#!/usr/bin/env python3
"""
Prueba r√°pida para verificar que protobuf funciona
"""

import sys
import os

# Agregar ruta actual al PYTHONPATH
sys.path.insert(0, os.getcwd())

print("üß™ PROBANDO IMPORTACIONES...")
print("=" * 40)

# Probar protobuf
try:
    from src.protocols.protobuf import network_event_pb2

    print("‚úÖ Protobuf importado correctamente")

    # Crear instancia de prueba
    event = network_event_pb2.NetworkEvent()
    event.event_id = "test_123"
    event.source_ip = "192.168.1.1"
    event.target_ip = "192.168.1.2"
    event.agent_id = "test-agent"
    print("‚úÖ NetworkEvent creado exitosamente")

    # Probar serializaci√≥n
    data = event.SerializeToString()
    print(f"‚úÖ Serializaci√≥n OK: {len(data)} bytes")

    # Probar deserializaci√≥n
    event2 = network_event_pb2.NetworkEvent()
    event2.ParseFromString(data)
    print(f"‚úÖ Deserializaci√≥n OK: event_id = {event2.event_id}")

except Exception as e:
    print(f"‚ùå Error con protobuf: {e}")
    import traceback

    traceback.print_exc()

# Probar scapy
try:
    from scapy.all import IP, TCP, UDP

    print("‚úÖ Scapy importado correctamente")
except Exception as e:
    print(f"‚ùå Error con scapy: {e}")

# Probar zmq
try:
    import zmq

    print("‚úÖ ZMQ importado correctamente")
except Exception as e:
    print(f"‚ùå Error con zmq: {e}")

print("=" * 40)
print("üéØ RESULTADO: ", end="")

# Verificar si todo est√° bien
try:
    from src.protocols.protobuf import network_event_pb2
    from scapy.all import IP, TCP
    import zmq

    print("‚úÖ TODAS LAS IMPORTACIONES FUNCIONAN")
    print("\\nüöÄ Puedes ejecutar:")
    print("   python agent_scapy_fixed.py")
    print("   python -m src.agents.agent_scapy")
except:
    print("‚ùå A√öN HAY PROBLEMAS")